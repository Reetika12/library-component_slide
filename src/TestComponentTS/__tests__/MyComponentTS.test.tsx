import * as React from 'react';
import { act } from 'react-dom/test-utils';
import * as ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import { TestComponentTS } from '../index';

describe('MyComponentTS', function test() {
	it('should render correctly', () => {
		const component = shallow(<TestComponentTS title="1" />);

		expect(component).toMatchSnapshot();
	});

	it('should display title passed as props', function () {
		const title = 'hello';
		const container = document.createElement('div');
		document.body.appendChild(container);
		act(() => {
			ReactDOM.render(<TestComponentTS title={title} />, container);
		});
		const header = container.querySelector('h1');
		expect(header?.textContent).toBe(title);
	});

	it('should display subtitle passed as props', function () {
		const title = 'hello';
		const subtitle = 'world';
		const container = document.createElement('div');
		document.body.appendChild(container);
		act(() => {
			ReactDOM.render(<TestComponentTS title={title} subTitle={subtitle} />, container);
		});
		const header = container.querySelector('h2');
		expect(header?.textContent).toBe(subtitle);
	});
});
