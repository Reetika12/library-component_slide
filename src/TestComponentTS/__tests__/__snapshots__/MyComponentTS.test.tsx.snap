// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponentTS should render correctly 1`] = `
<div
  className="test-component-ts"
>
  <h1>
    1
  </h1>
  <h2>
    World
  </h2>
  <p>
    This is an e1xample component of Typescript component
  </p>
</div>
`;
